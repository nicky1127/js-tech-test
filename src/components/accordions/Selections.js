import React, { useState } from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import { connect } from "react-redux";
import { makeStyles } from "@material-ui/core/styles";
import {
  Accordion,
  AccordionSummary,
  AccordionDetails,
  Typography,
  Box
} from "@material-ui/core";
import { ExpandMore, Clear } from "@material-ui/icons";
import SelectionUnit from "components/units/SelectionUnit";
import { getEventById, removeAllSelection } from "redux/_actions";
import labels from "constants/labels";

const constants = labels.Selections;

const useStyles = makeStyles(theme => ({
  root: {
    width: "95%",
    maxHeight: "95%",
    overflowY: "scroll",
    margin: "10px auto"
  },
  header: {
    color: "#fff",
    backgroundColor: "rgb(10,40,108)",
    borderTopLeftRadius: "5px",
    borderTopRightRadius: "5px"
  },
  icon: { color: "#fff" },
  details: {
    padding: 0,
    flexDirection: "column"
  },
  //remove all selections
  rmAll: {
    marginTop: "10px",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    fontWeight: "500",
    width: "60%",
    fontSize: "12px",
    height: "30px",
    padding: "4px",
    border: "1px solid rgb(179,191,211)",
    borderRadius: "5px",
    "&:hover": {
      backgroundColor: "rgb(239,244,252)",
      cursor: "pointer"
    }
  }
}));

export const Selections = props => {
  const classes = useStyles();
  const { selections, removeAllSelection } = props;
  const [expanded, setExpanded] = useState(true);

  let content;
  if (selections.length > 0) {
    content = selections.map(selection => (
      <SelectionUnit outcome={selection} />
    ));
  }

  const onClickRemoveAll = () => {
    removeAllSelection();
  };

  return (
    <div className={clsx("Selections", classes.root)} data-testid="selections">
      <Accordion expanded={expanded}>
        <AccordionSummary
          className={classes.header}
          onClick={() => setExpanded(!expanded)}
          expandIcon={<ExpandMore fontSize="small" className={classes.icon} />}
          aria-controls="panel1a-content"
          id="panel1a-header"
        >
          <Typography className={classes.heading}>{`Singles`}</Typography>
        </AccordionSummary>
        <AccordionDetails classes={{ root: classes.details }}>
          {content}
        </AccordionDetails>
      </Accordion>
      <Box className={classes.rmAll} onClick={onClickRemoveAll}>
        <Clear fontSize="small" />
        Remove all selections
      </Box>
    </div>
  );
};

Selections.defaultProps = { selections: [] };

Selections.propTypes = { selections: PropTypes.array };

const mapStateToProps = state => {
  const { selections, isOddsDecimal } = state;
  return { selections, isOddsDecimal };
};

const ConnectedSelections = connect(mapStateToProps, {
  getEventById,
  removeAllSelection
})(Selections);

export default ConnectedSelections;
