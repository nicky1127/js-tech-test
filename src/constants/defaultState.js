const state = {
  liveEvents: [],
  event: {},
  markets: {},
  isOddsDecimal: false,
  loading: false,
  error: {},
  selections: []
};

export default state;
