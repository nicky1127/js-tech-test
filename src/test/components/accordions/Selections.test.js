import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Selections from "components/accordions/Selections";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import defaultState from "constants/defaultState";

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});

const props = {};

const state = {
  ...defaultState
};
const mockStore = configureMockStore();
const store = mockStore(state);

describe("Selections", () => {
  it("Snapshot matchs", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Selections {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Selections should render one [data-testid=selections]", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Selections {...props} />
      </Provider>
    );
    const container = getByTestId("selections");
    expect(container).toBeInTheDocument();
  });
});
