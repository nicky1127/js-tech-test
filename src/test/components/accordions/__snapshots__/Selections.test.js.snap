// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selections Snapshot matchs 1`] = `
<div
  className="Selections makeStyles-root-1"
  data-testid="selections"
>
  <div
    className="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-controls="panel1a-content"
      aria-disabled={false}
      aria-expanded={true}
      className="MuiButtonBase-root MuiAccordionSummary-root makeStyles-header-2 Mui-expanded"
      id="panel1a-header"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content Mui-expanded"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Singles
        </p>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-3 MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-entered"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel1a-header"
            id="panel1a-content"
            role="region"
          >
            <div
              className="MuiAccordionDetails-root makeStyles-details-4"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-6 makeStyles-rmAll-5"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
    Remove all selections
  </div>
</div>
`;
